python/cudf/cudf/pylibcudf_tests/test_binaryops.py

```diff
@@ -114,90 +110,111 @@ def get_result():
     ids=idfn,
 )
 def test_add(pa_data, plc_data):
-    def add(x, y):
-        if x is None or y is None:
-            return None
-        return x + y
-
     _test_binaryop_inner(
         pa_data,
         plc_data,
-        add,
+        pa.compute.add,
         plc.binaryop.BinaryOperator.ADD,
     )
 
 
 @pytest.mark.parametrize(
     "pa_data",
-    [("int64", "int64", "int64"), ("int64", "float64", "float64")],
+    [
+        ("int64", "int64", "int64"),
+        ("int64", "float64", "float64"),
+        ("int64", "int64", "datetime64[ns]"),
+    ],
     indirect=True,
     ids=idfn,
 )
 def test_sub(pa_data, plc_data):
-    def sub(x, y):
-        if x is None or y is None:
-            return None
-        return x - y
-
     _test_binaryop_inner(
         pa_data,
         plc_data,
-        sub,
+        pa.compute.subtract,
```

Comment:
It looks like the only difference in all of these tests is the pair of libcudf-side BinaryOperator and the matching function for doing the compute on the host.

So, can we have a fixture that returns the pair of functions and then promote `_test_binaryop_inner` to the top-level?

If not, can you explain why that wouldn't work?

Reply:
Oh, it's the same issue as the indirect fixtures? you need a different set of fixtures for different classes of binops.